lib/Runtime/Library/JavascriptRegExpConstructor.cpp

```diff
@@ -53,17 +54,44 @@ namespace Js
         this->lastInput = lastInput;
         this->lastMatch = lastMatch;
         this->reset = true;
+        this->invalidatedLastMatch = false;
+    }
+
+    void JavascriptRegExpConstructor::InvalidateLastMatch(UnifiedRegex::RegexPattern* lastPattern, JavascriptString* lastInput)
+    {
+        AssertMsg(lastPattern != nullptr, "lastPattern should not be null");
+        AssertMsg(lastInput != nullptr, "lastInput should not be null");
+        AssertMsg(JavascriptOperators::GetTypeId(lastInput) != TypeIds_Null, "lastInput should not be JavaScript null");
+
+        this->lastPattern = lastPattern;
+        this->lastInput = lastInput;
+        this->lastMatch.Reset();
+        this->reset = true;
+        this->invalidatedLastMatch = true;
     }
 
     void JavascriptRegExpConstructor::EnsureValues()
     {
         if (reset)
         {
-            Assert(!lastMatch.IsUndefined());
             ScriptContext* scriptContext = this->GetScriptContext();
+            const CharCount lastInputLen = lastInput->GetLength();
+            const wchar_t* lastInputStr = lastInput->GetString();
```

Comment:
use `WCHAR` instead. It refers to `char16` on xplat. (why? wchar_t is 4 bytes on some platforms while 2 bytes on others)

Reply:
I replaced it with `char16` since all the APIs here are using `char16`. Seems this must be fine for xplat.

Reply:
That is also fine! Thanks